packages/fork-choice/src/forkChoice/forkChoice.ts

```diff
@@ -943,6 +943,21 @@ export class ForkChoice implements IForkChoice {
     return this.protoArray.getAllNonAncestorNodes(blockRoot);
   }
 
+  /**
+   * Returns both ancestor and non-ancestor blocks in a single traversal.
+   */
+  getAllAncestorAndNonAncestorBlocks(blockRoot: RootHex): {
```

Comment:
please add a unit test to make sure `getAllAncestorAndNonAncestorBlocks = getAllAncestorNodes + getAllNonAncestorBlocks`